### PR TITLE
Adding event-based likelihood evaluation

### PR DESCRIPTION
**Pull request by ss15859**
Added a class 'ETASLikelihoodCalculation', which takes the inverted parameters and calculates the event-based log-likelihood of events in a testing window.

The testing window is a subset of the catalog that is not shown to ETASParameterCalculation.

log-likelihood is decomposed into spatial and temporal log-likelihood and is compared with a homogeneous Poisson baseline.

### CODE REVIEW

File: etas/evaluation.py [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+
+def to_days(timediff):
+    return timediff / np.timedelta64(1, 'D')
+
+
+
+class ETASLikelihoodCalculation(ETASParameterCalculation):
+    def __init__(self, metadata: dict):
+        """
+        Class to invert ETAS parameters.
```
Comment (lmizrahi) [inline etas/evaluation.py:37]: shouldn't this be the docstring describing the new class? it's ok not to do it now, but we can adjust it at some point

File: etas/evaluation.py [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+                haversine(
+                    lat_rads[i],
+                    lat_rads[:i],
+                    long_rads[i],
+                    long_rads[:i],
+                    earth_radius,
+                )
+            )
+
+def to_days(timediff):
```
Comment (lmizrahi) [inline etas/evaluation.py:29]: could probably import the function from inversion.py? or not?

**Review by lmizrahi — APPROVED** [review]
Looks good! I just added a few comments that we don't need to do now, but let's keep them in mind. Maybe I create an issue so that we/I remember to look at these at some point

File: etas/evaluation.py (hunk 2 above)
Comment (ss15859) [reply etas/evaluation.py:29]: You're right, although this one in evaluation.py uses numpy timedelta64, whereas the one in inversion.py uses pandas dt.timedelta.

The repeat definition should be avoided nonetheless.

**Pull request merged**